### PR TITLE
Change BLE appearance setting to "smart watch"

### PR DESCRIPTION
**Pull request by evergreen22**
Set the appearance value to 'smart watch' as documented in the bluetooth specification.

https://specificationrefs.bluetooth.com/assigned-values/Appearance%20Values.pdf

### CODE REVIEW

**Comment by Riksu9000** [comment]
The document says 0xC2 is Smartwatch?

**Comment by evergreen22** [comment]
On Sun, 29 Aug 2021, Riku Isokoski wrote:

> The document says 0xC2 is Smartwatch?
>

Yes, the document lists bluetooth/ble code+subcodes for appearance. The 
ble appearance for watch + smart watch is 0xC2.

https://specificationrefs.bluetooth.com/assigned-values/Appearance%20Values.pdf

Before this patch pinetime showed up as 'unknown'. After this patch it 
shows up as 'smart watch'.

**Comment by evergreen22** [comment]
On Sun, 29 Aug 2021, Riku Isokoski wrote:

> The document says 0xC2 is Smartwatch?
>

Ah yes... I see you were being subtle. The doc does say 0xC2 and I fat 
fingered the PR.

Fixed now.

Thanks.

File: src/components/ble/NimbleController.cpp [hunk 1]
```diff
@@ -71,7 +71,7 @@ void NimbleController::Init() {
   ASSERT(res == 0);
   res = ble_svc_gap_device_name_set(deviceName);
   ASSERT(res == 0);
-  rc = ble_svc_gap_device_appearance_set(0xC4);
+  rc = ble_svc_gap_device_appearance_set(0xC2);
```
Comment (hubmartin) [inline src/components/ble/NimbleController.cpp:74]: Not sure if I cloned it right, but the variable `rc` does not exist in that function. I had to change it to `res` variable whic is used to check return values from BLE functions.
```
/home/martin/Desktop/dev/watches/InfiniTime/src/components/ble/NimbleController.cpp: In member function 'void Pinetime::Controllers::NimbleController::Init()':
/home/martin/Desktop/dev/watches/InfiniTime/src/components/ble/NimbleController.cpp:74:3: error: 'rc' was not declared in this scope; did you mean 'erfc'?
   74 |   rc = ble_svc_gap_device_appearance_set(0xC2);
      |   ^~
```

**Comment by evergreen22** [comment]
Of course you are right.

I pulled this out of a local-only WIP branch where I am refactoring the 
advertising logic and _obviously_ did not a good job of proofing the PR.

I apologize to you and riksu for the trouble. It is now fixed.

**Comment by hubmartin** [comment]
No troubles :)
How I can check that icon have changed? I have Android and it doesn't seem to display any icon. Just BLE logo in the system scan, the same is in NRFConnect. Thanks.

**Comment by evergreen22** [comment]
In NRFConnect you can see the value for appearance changed from zero 
'unknown' to 194 'smart watch'.

**Comment by evergreen22** [comment]
NRFConnect say '[194] Unknown' with the appearance set to 0xC2.

It says '[192] Generic Watch (Generic category)' with the appearance 
set to 0xC0.

**Comment by geekbozu** [comment]
Smart watch is a sub category, may just not be implemented right. Either way simple enough and no reason to not properly identify ourselves. However do you need to call it twice? It should not change after init as its just setting a global variable in the svc_gap scope, and nothing else ever writes to it.

**Comment by evergreen22** [comment]
On Sun, 29 Aug 2021, Tim Keller wrote:

> Smart watch is a sub category, may just not be implemented right. Either 
> way simple enough and no reason to not properly identify ourselves. 
> However do you need to call it twice? It should not change after init as 
> its just setting a global variable in the svc_gap scope, and nothing 
> else ever writes to it.
>
>

No, setting the name and appearance should only be done once and it does 
not need to be repeated in advertising. I address this in my WIP PR #625.

I was trying to present a one PR with a simple change for appearance. I 
clearly failed by not adequately proofing the original PR. :-(

**Comment by geekbozu** [comment]
Don't feel off this is why we do reviews! I would change this PR to only have it in init, Then we can mark this one as good for merge!

**Comment by evergreen22** [comment]
On Sun, 29 Aug 2021, Tim Keller wrote:

> Don't feel off this is why we do reviews! I would change this PR to only 
> have it in init, Then we can mark this one as good for merge!
>
>

In PR #625 lines 90-93 of Init is where I'm setting them both in my local 
branch. I'll update this PR with that code.

File: src/components/ble/NimbleController.cpp (hunk 1 above)
Comment (evergreen22) [reply src/components/ble/NimbleController.cpp]: The device name and appearance are not set properly in Init as discussed in the comments.

**Comment by Riksu9000** [comment]
#625 was merged which included this change so this can be closed.

**Comment by evergreen22** [comment]
Closed by #625

**Pull request closed without merging**